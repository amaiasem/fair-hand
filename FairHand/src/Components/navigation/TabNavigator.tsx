import React from 'react'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import Home from '../../screens/Home/Home'
import Maps from '../../screens/Map/Map'
import User from '../../screens/User/User'
import TabBar from './TabBar'
import Screen from '../../../constants/ScreenConstants'

const Tab = createBottomTabNavigator()

const TabNavigator = () => {
  return (
    <Tab.Navigator
    tabBar={(props) => <TabBar {...props}/>}
    initialRouteName={Screen.Home}
    >
        <Tab.Screen
          name={Screen.Home}
          component={Home}
          initialParams={{ icon: 'home' }}
          />
        <Tab.Screen
        name={Screen.Maps}
        component={Maps}
        initialParams={{ icon: 'enviromento' }}
        />
        <Tab.Screen
        name={Screen.User}
        component={User}
        initialParams={{ icon: 'user' }}
        />
    </Tab.Navigator>
  )
}

export default TabNavigator
