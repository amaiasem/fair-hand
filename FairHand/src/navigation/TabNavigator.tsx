import React from 'react'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import Home from '../screens/Home'
import Maps from '../screens/Map'
import User from '../screens/User'
import TabBar from './TabBar'

const Tab = createBottomTabNavigator()

const TabNavigator = () => {
  return (
    <Tab.Navigator
    tabBar={(props) => <TabBar {...props}/>}
    initialRouteName={'Home'}
    >
        <Tab.Screen
          name='Home'
          component={Home}
          initialParams={{ icon: 'home' }}
          />
        <Tab.Screen
        name='Maps'
        component={Maps}
        initialParams={{ icon: 'enviromento' }}
        />
        <Tab.Screen
        name='Shop'
        component={User}
        initialParams={{ icon: 'user' }}
        />
    </Tab.Navigator>
  )
}

export default TabNavigator
