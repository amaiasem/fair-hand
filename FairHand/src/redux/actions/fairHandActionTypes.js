export default {
  LOAD_ALL_SHOPS: 'LOAD_ALL_SHOPS'
}
