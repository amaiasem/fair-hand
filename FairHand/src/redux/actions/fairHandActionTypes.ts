export default {
  LOAD_ALL_SHOPS: 'LOAD_ALL_SHOPS',
  FILTER_SHOP_BY_TYPE: 'FILTER_SHOP_BY_TYPE',
  FILTER_SHOP_BY_NAME: 'FILTER_SHOP_BY_NAME',
  GET_REVIEWS_BY_SHOP_NAME: 'GET_REVIEWS_BT_SHOP_NAME',
  GET_REVIEWS_BY_USER_NAME: 'GET_REVIEWS_BY_USER_NAME',
  USER_SIGN_IN: 'USER_SIGN_IN',
  USER_REGISTER: 'USER_REGISTER',
  USER_SIGN_OUT: 'USER_SIGN_OUT',
  ADD_REVIEW: 'ADD_REVIEW'
}
