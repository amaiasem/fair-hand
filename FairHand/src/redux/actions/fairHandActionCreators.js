import axios from 'axios'
import fairHandActionTypes from './fairHandActionTypes'

export default function loadAllShops () {
  return async function fetchInfo (dispatch) {
    const { data } = await axios('http://192.168.0.41:5000/shops')
    dispatch({
      type: fairHandActionTypes.LOAD_ALL_SHOPS,
      data
    })
  }
}

export function filterShopsByType (shops, typeClothes) {
  const filteredShops = shops.filter(shop => shop.type.includes(typeClothes) === true)

  return {
    type: fairHandActionTypes.FILTER_SHOP_BY_TYPE,
    data: filteredShops
  }
}

export function filterShopsByName (shops, shopName) {
  const filteredShops = shops.filter(shop => shop.shopName.includes(shopName) === true)

  return {
    type: fairHandActionTypes.FILTER_SHOP_BY_NAME,
    data: filteredShops
  }
}
