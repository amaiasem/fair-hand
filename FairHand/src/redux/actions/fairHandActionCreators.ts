import axios from 'axios'
import fairHandActionTypes from './fairHandActionTypes'
import ShopInterface from '../../Interfaces/shopInterface'
import Review from '../../Interfaces/reviewInterface'

// Shop actions

export function loadAllShops () {
  return async function fetchInfo (dispatch: any) {
    const { data } = await axios.get('http://192.168.0.41:5000/shops')
    dispatch({
      type: fairHandActionTypes.LOAD_ALL_SHOPS,
      data
    })
  }
}

export function filterShopsByType (shops: ShopInterface[], typeClothes: string) {
  const filteredShops = shops.filter(shop => shop.type.includes(typeClothes) === true)

  return {
    type: fairHandActionTypes.FILTER_SHOP_BY_TYPE,
    data: filteredShops
  }
}

export function filterShopsByName (shops: ShopInterface[], shopName: string) {
  if (shopName === '') {
    return {
      type: fairHandActionTypes.FILTER_SHOP_BY_NAME,
      data: shops
    }
  } else {
    const filteredShops = shops.filter(shop => shop.shopName.includes(shopName) === true)

    return {
      type: fairHandActionTypes.FILTER_SHOP_BY_NAME,
      data: filteredShops
    }
  }
}

// Reviews actions

export function getReviewsByShopName (searchShop:string) {
  return async function fetchInfo (dispatch: any) {
    const { data } = await axios.get(`http://192.168.0.41:5000/reviews/shopName/${searchShop}`)

    dispatch({
      type: fairHandActionTypes.GET_REVIEWS_BY_SHOP_NAME,
      data
    })
  }
}

export function getReviewsByUserName (searchName:string) {
  return async function fetchInfo (dispatch: any) {
    const { data } = await axios.get(`http://192.168.0.41:5000/reviews/userName/${searchName}`)

    dispatch({
      type: fairHandActionTypes.GET_REVIEWS_BY_USER_NAME,
      data
    })
  }
}

// User actions

export function userLogin (user: object) {
  return async (dispatch: any) => {
    try {
      const { data } = await axios.post('http://192.168.0.41:5000/auth/login', user)
      dispatch({
        type: fairHandActionTypes.USER_SIGN_IN,
        data
      })
    } catch (error) {
      dispatch({
        type: fairHandActionTypes.USER_SIGN_IN,
        data: 400
      })
    }
  }
}

export function userRegister (user: object) {
  return async (dispatch: any) => {
    const { data } = await axios.post('http://192.168.0.41:5000/auth/register', user)
    if (data === 'User already exists!') {
      dispatch({
        type: fairHandActionTypes.USER_REGISTER
      })
    } else {
      dispatch({
        type: fairHandActionTypes.USER_REGISTER,
        data
      })
    }
  }
}

export function addReview (review: Review) {
  return async (dispatch: any) => {
    try {
      const { data } = await axios.post('http://192.168.0.41:5000/reviews', review)
      dispatch({
        type: fairHandActionTypes.ADD_REVIEW,
        data
      })
    } catch (error) {
      dispatch({
        type: fairHandActionTypes.ADD_REVIEW,
        data: 400
      })
    }
  }
}

export function deleteReview (allReviews: Review[], reviewID: Review) {
  return async (dispatch: any) => {
    try {
      const { data } = await axios.delete('http://192.168.0.41:5000/reviews', {
        data: {
          _id: reviewID
        }
      })
      dispatch({
        type: fairHandActionTypes.DELETE_REVIEW,
        data: { allReviews: allReviews, deleted: data }
      })
    } catch (error) {
      dispatch({
        type: fairHandActionTypes.DELETE_REVIEW,
        data: 400
      })
    }
  }
}

export function userUpdate (user: object) {
  return async (dispatch: any) => {
    const { data } = await axios.put('http://192.168.0.41:5000/user', user)
    if (data === 'Could not update the user') {
      dispatch({
        type: fairHandActionTypes.USER_UPDATE
      })
    } else {
      dispatch({
        type: fairHandActionTypes.USER_UPDATE,
        data
      })
    }
  }
}
