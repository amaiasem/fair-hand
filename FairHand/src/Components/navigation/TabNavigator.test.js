import React from 'react'
import { Provider } from 'react-redux'
import { render, waitFor } from '@testing-library/react-native'
import TabNavigator from './TabNavigator.tsx'
import configureStore from 'redux-mock-store'
import { NavigationContainer } from '@react-navigation/native'
import * as action from '../../redux/actions/fairHandActionCreators'
jest.mock('../../redux/actions/fairHandActionCreators')

const mockStore = configureStore([])
jest.spyOn(action, 'loadAllShops').mockReturnValue({ type: '' })

describe('Given a component TabNavigator', () => {
  let store
  beforeEach(() => {
    store = mockStore({
      shopReducer: { shops: [], shop: {}, filteredShops: [] }
    })
  })

  it('renders the correct screen', async () => {
    const { getByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <TabNavigator />
            </NavigationContainer>
        </Provider>
    )
    await waitFor(() => getByText('Maps'))
  })
})
