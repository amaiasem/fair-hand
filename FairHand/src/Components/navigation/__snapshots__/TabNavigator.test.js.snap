// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a component TabNavigator renders the correct screen 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ForwardRef(NavigationContainer)>
    <TabNavigator />
  </ForwardRef(NavigationContainer)>
</Provider>
`;
